lib/api/funnel.js

```diff
@@ -368,9 +370,37 @@ class Funnel {
    * @returns {Promise<Request>}
    */
   async checkRights (request) {
+   
+    let skipTokenVerification = false;
+    // When the Support of Cookie as Authentication Token is enabled we check if an auth token cookie is present
+    // When a request is made with CookieOnly set to true
+    // We try to use the auth token cookie if present as auth token
+    // otherwise check for auth token as input
+    if ( global.kuzzle.config.http.supportCookieAuthentication
+      && request.input.args.cookieOnly
+      && request.input.headers
+      && has(request.input.headers, 'cookie')
+    ) {
+      try {
+        const cookie = Cookie.parse(request.input.headers.cookie);
+
+        request.input.jwt = cookie.authToken || null;
+        
+        skipTokenVerification = request.input.controller === 'auth'
+          && (request.input.action === 'login' || request.input.action === 'checkToken');
+      } catch {
+        throw kerror.get(
+          'security',
+          'cookie',
+          'invalid'
+        );
```

Comment:
try..catch should be specialised as much as possible, for better maintenability: this helps maintainers knowing what exactly is caught and why

```suggestion
      let cookie;
      
      try {
        cookie = Cookie.parse(request.input.headers.cookie);
      } 
      catch {
        throw kerror.get('security', 'cookie', 'invalid');
      }
        
      request.input.jwt = cookie.authToken || null;
        
      skipTokenVerification = request.input.controller === 'auth'
        && (request.input.action === 'login' || request.input.action === 'checkToken');
```